### PR TITLE
adding random_state to mnist

### PR DESCRIPTION
**Pull request by kmader**
adding random state to mnist will make the results more reproducible

### CODE REVIEW

**Comment by danielegrattarola** [comment]
Thanks for the PR!
Is this different than calling `np.random.seed` before `load_data`?
Because, if so, I might need to update all dataset loaders based on sklearn to match the same signature.

**Comment by kmader** [comment]
@danielegrattarola it's most important for the `train_test_split` since those can directly affect the training, the other part is the same as having it before.
